File: 351003/Naskevich/src/repositories/marker.py
from abc import ABC, abstractmethod

from src.models.marker import Marker


class AbstractMarkerRepository(ABC):

    @abstractmethod
    async def get_by_id(self, entity_id: int) -> Marker | None:
        raise NotImplementedError

    @abstractmethod
    async def get_all(self) -> list[Marker]:
        raise NotImplementedError

    @abstractmethod
    async def create(self, entity: Marker) -> Marker:
        raise NotImplementedError

    @abstractmethod
    async def update(self, entity: Marker) -> Marker | None:
        raise NotImplementedError

    @abstractmethod
    async def delete(self, entity_id: int) -> bool:
        raise NotImplementedError

    @abstractmethod
    async def get_by_name(self, name: str) -> Marker | None:
        raise NotImplementedError 

File: 351001/Radzetskii/news-board/publisher/internal/domain/models/marker.go
package models

import "context"

type Marker struct {
	ID   int64
	Name string
}

type MarkerRepository interface {
	Create(ctx context.Context, marker *Marker) error
	GetByID(ctx context.Context, id int64) (*Marker, error)
	GetByName(ctx context.Context, name string) (*Marker, error)
	GetAll(ctx context.Context, limit, offset int) ([]Marker, error)
	Update(ctx context.Context, marker *Marker) (bool, error)
	Delete(ctx context.Context, id int64) (bool, error)
	GetByNewsID(ctx context.Context, newsID int64) ([]Marker, error)
	CountNewsByMarker(ctx context.Context, markerID int64) (int, error)
}
